security-framework/src/certificate.rs

```diff
@@ -57,8 +86,152 @@ impl SecCertificate {
             CFString::wrap_under_create_rule(summary).to_string()
         }
     }
+
+    #[cfg(any(feature = "OSX_10_12", target_os = "ios"))]
+    /// Returns DER encoded subjectPublicKeyInfo of certificate if available. This can be used
+    /// for certificate pinning.
+    pub fn public_key_info_der(&self) -> Result<Option<Vec<u8>>> {
```

Comment:
My concern is that this is a very specific single-purpose high-level function. Could it be broken down into components that help others write their `public_key_info_der` and alike themselves?